### ISSUE TITLE
@ManyToMany return all rows in table

### ISSUE THREAD

**Issue by Fokir** [typeorm/typeorm#223]
I have model
```
@Table()
export class User extends Model{
    @PrimaryGeneratedColumn()
    id: number;
    ...
    @ManyToMany(type=>Role, (role:Role)=>role.users)
    roles: Promise<Array<Role>>;
    ...
}
```
and 
```
@Table()
export class Role extends Model{
    static __repository: Repository<Role>;

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    @Index({unique: true})
    role: string;

    @ManyToMany(type => User, (user: User) => user.roles)
    @JoinTable()
    users: Promise<Array<User>>;
}
```

![table](https://cloud.githubusercontent.com/assets/1648384/22175236/87a4b53c-e00a-11e6-84fb-45fd2c50995e.png)

```
let user: User = await token.user;
user.roles.then((d)=>{
    console.log(d);
});
```
Returns all roles from the role of the table, ignoring the staging table?
 I'm use 0.0.7-alpha.22
 
![unknown 1](https://cloud.githubusercontent.com/assets/1648384/22175285/6eef78dc-e00b-11e6-8859-9d81eff04999.png)

**Comment by Fokir** [comment]
Can I wait for error correction in coming days?

**Comment by ItAlexander** [comment]
+

**Comment by SetgorS** [comment]
+

**Comment by pleerock** [comment]
haha you have a strong team 😆 . okay I'll make a fix for this this week.

**Comment by Fokir** [comment]
Thank you, it's just an important place in the application. A "typeorm" has great potential in applications Typescript

**Comment by Fokir** [comment]
The problem is fixed?

**Comment by pleerock** [comment]
Yes. released in `0.0.8-alpha.2`
